src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs

```diff
@@ -58,25 +58,28 @@ public AgentHealthReporter(IMetricBuilder metricBuilder, IScheduler scheduler)
         public override void Dispose()
         {
             base.Dispose();
-            _scheduler.StopExecuting(LogRecurringLogs);
+            _scheduler.StopExecuting(LogPeriodicReport);
         }
 
-        private void LogRecurringLogs()
+        private void LogPeriodicReport()
         {
             foreach (var data in _recurringLogDatas)
             {
                 data?.LogAction(data.Message);
             }
-
+            
+            List<string> events = new List<string>();
             foreach (var counter in _agentHealthEventCounters)
             {
                 if (counter.Value != null && counter.Value.Value > 0)
                 {
                     var agentHealthEvent = counter.Key;
                     var timesOccured = counter.Value.Exchange(0);
-                    Log.Info($"Event {agentHealthEvent} has occurred {timesOccured} times in the last {_timeBetweenExecutions.TotalSeconds} seconds");
+                    events.Add(string.Format("{0} {1} {2}", timesOccured, agentHealthEvent, (timesOccured == 1) ? "event" : "events"));
```

Comment:
Totally a nitpick, but `string.Format()` went out of vogue years ago... ;-)

Reply:
Are you talking about string interpolation or is there some other formatter I'm not thinking of? I don't like using interpolation with conditionals, I think it's harder to read. But maybe that's just me. :)

Reply:
Yeah, string interpolation is where I was headed. Pretty easy to write this one as 
```
events.Add($"{timesOccured} {agentHealthEvent}, {(timesOccured == 1 ? "event" : "events")}
```
The trick being to wrap the ternary in parentheses.

Oh - one more nit: `timesOccured` should be `timesOccurred` ...

Reply:
If all the things your are putting into the string are strings, the performance is close to String.Format, but if they are not strings, older versions of .NET have issues with the new interpolation style.  Calling ToString on the non-strings resolves it however.